### PR TITLE
[abnormal_security] Update data-collection to handle empty threats

### PR DESCRIPTION
**Pull request by brijesh-elastic**
Type of change
- Bug

## Proposed commit message

When we receive empty threat data, CEL stores the `worklist` as `{"total": 0, "threats": [], "pageNumber": 1}` instead of an empty object `{}`. As a result, in the next interval, the condition `size(state.worklist) > 0` evaluates to true, preventing the CEL from making REST calls to the `/threats` endpoint and causing it to get stuck in a loop until it is restarted.

To address this issue, we updated the condition to `size(state.worklist.threats) > 0` for the `threat` data stream, ensuring that empty threat arrays are handled correctly and REST calls continue during subsequent intervals.

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

- Clone integrations repo.
- Install elastic package locally.
- Start elastic stack using elastic-package.
- Move to integrations/packages/abnormal_security directory.
- Run the following command to run tests.
> elastic-package test

### CODE REVIEW

File: packages/abnormal_security/data_stream/threat/agent/stream/cel.yml.hbs [hunk 1]
```diff
@@ -25,7 +25,7 @@ redact:
     - access_token
 program: |
   (
-    has(state.worklist) && size(state.worklist) > 0 ?
+    has(state.worklist) && has(state.worklist.threats) && size(state.worklist.threats) > 0 ?
```
Comment (efd6) [inline packages/abnormal_security/data_stream/threat/agent/stream/cel.yml.hbs:28]:
```suggestion
    has(state.?worklist.threats) && size(state.worklist.threats) > 0 ?
```

**Review by efd6 — COMMENTED** [review]
Is there an issue associated with this?

**Comment by brijesh-elastic** [comment]
> Is there an issue associated with this?

It was reported via slack.

**Review by efd6 — APPROVED** [review]
Thanks

**Pull request merged**